### PR TITLE
feat(mdn): init

### PR DESCRIPTION
**Pull request by soya-daizu**
## 🎉 Theme for MDN 🎉

MDN Web Docs is an open-source, collaborative project documenting Web platform technologies, including [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS), [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML), [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript), and [Web APIs](https://developer.mozilla.org/en-US/docs/Web/API/).

Screenshot 2023-12-12 at 14-51-17 MDN Web Docs

Screenshot 2023-12-12 at 14-51-41 MDN Web Docs

Screenshot 2023-12-12 at 15-00-11 @layer - CSS Cascading Style Sheets MDN

Screenshot 2023-12-12 at 15-00-19 @layer - CSS Cascading Style Sheets MDN

Few other examples:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static

https://developer.mozilla.org/en-US/docs/Web/CSS/@layer

https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/isContextLost

## 💬 Additional Comments 💬

The linter is complaining for not using complex `:not()` notation for `:root` pseudo-class selector, but since the original style is using a chain of `:not()` notation, we have to stick with that to make the styles work.

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/src/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - `catppuccin.user.css` - all the CSS for the userstyle, based on the
    template.
  - `assets/mocha.webp`, `assets/macchiato.webp`, `assets/frappe.webp` and
    `assets/latte.webp` - individual screenshots of the themed website, in all 4
    Catppuccin flavors.
  - `catwalk.webp` - image of all individual screenshots stitched together,
    generated via [catwalk](https://github.com/catppuccin/toolbox#-catwalk).

### CODE REVIEW

**Comment by uncenter** [comment]
Can you fix the other lint issues (function-url-quotes)?

File: styles/mdn/catppuccin.user.css [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+@namespace      github.com/catppuccin/userstyles/styles/mdn
+@homepageURL    https://github.com/catppuccin/userstyles/tree/main/styles/mdn
+@version        0.1.0
+@description    Soothing pastel theme for MDN
+@author         Catppuccin
+@updateURL      https://github.com/catppuccin/userstyles/raw/main/styles/mdn/catppuccin.user.css
+@license        MIT
+
+@preprocessor   less
+@var select lightFlavor "Light Flavor" ["latte:Latte", "frappe:Frappe", "macchiato:Macchiato", "mocha:Mocha*"]
```
Comment (uncenter) [inline styles/mdn/catppuccin.user.css:12]: Not sure how this got changed from the template but the default light flavor needs to be Latte.
```suggestion
@var select lightFlavor "Light Flavor" ["latte:Latte*", "frappe:Frappe", "macchiato:Macchiato", "mocha:Mocha*"]
```

File: styles/mdn/catppuccin.user.css [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  --icon-primary: @text;
+  --icon-secondary: @subtext0;
+  --icon-information: @blue;
+  --icon-warning: @peach;
+  --icon-critical: @maroon;
+  --icon-success: @green;
+  --accent-primary: @accent-Color;
+  --accent-primary-engage: fadeout(@accent-Color, 90%);
+  --accent-secondary: @accent-Color;
+  --accent-tertiary: fadeout(saturate(@accent-Color, 10%), 90%);
```
Comment (uncenter) [inline styles/mdn/catppuccin.user.css:93]: The `c` in `accent-Color` should be lowercase (`accent-color`).

```suggestion
  @accent-color: @catppuccin[@@lookup][@@accent];

  --text-primary: @text;
  --text-secondary: @subtext0;
  --text-active: @surface1;
  --text-inactive: fadeout(@subtext0, 40%);
  --text-link: @accent-color;
  --text-visited: lighten(saturate(@accent-color, -20%), -20%);
  --text-invert: @base;
  --background-primary: @base;
  --background-secondary: @crust;
  --background-tertiary: @mantle;
  --background-toc-active: @crust;
  --background-mark-yellow: fadeout(@yellow, 60%);
  --background-mark-green: fadeout(@green, 60%);
  --background-information: fadeout(@blue, 90%);
  --background-warning: fadeout(@maroon, 90%);
  --background-critical: fadeout(@red, 90%);
  --background-success: fadeout(@green, 90%);
  --border-primary: @surface2;
  --border-secondary: @surface1;
  --button-primary-default: @text;
  --button-primary-hover: @subtext0;
  --button-primary-active: @overlay0;
  --button-primary-inactive: @text;
  --button-secondary-default: @surface2;
  --button-secondary-hover: @surface1;
  --button-secondary-active: @overlay0;
  --button-secondary-inactive: @surface2;
  --button-secondary-border-focus: @blue;
  --button-secondary-border-red: @red;
  --button-secondary-border-red-focus: @maroon;
  --icon-primary: @text;
  --icon-secondary: @subtext0;
  --icon-information: @blue;
  --icon-warning: @peach;
  --icon-critical: @maroon;
  --icon-success: @green;
  --accent-primary: @accent-color;
  --accent-primary-engage: fadeout(@accent-color, 90%);
  --accent-secondary: @accent-color;
  --accent-tertiary: fadeout(saturate(@accent-color, 10%), 90%);
```

File: styles/mdn/catppuccin.user.css [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+  --baseline-high-check: @green;
+  --baseline-low-bg: hsl(hue(@blue), saturation(@blue), lightness(@surface0));
+  --baseline-low-engine-bg: hsl(hue(@blue), saturation(@blue), lightness(@base));
+  --baseline-low-check: @blue;
+  --baseline-low-img: url(/static/media/low-dark.7f95ec44f36ff865af95.svg);
+  --baseline-low-pill-bg: @sapphire;
+  --baseline-low-pill-color: @base;
+  --baseline-limited-bg: @surface0;
+  --baseline-limited-engine-bg: @base;
+  --baseline-limited-img: url(/static/media/limited-dark.2af083804a2999abf515.svg);
```
Comment (uncenter) [inline styles/mdn/catppuccin.user.css:189]: These look like MDN's default images right? Should we just get rid of these?
```suggestion
  --baseline-high-check: @green;
  --baseline-low-bg: hsl(hue(@blue), saturation(@blue), lightness(@surface0));
  --baseline-low-engine-bg: hsl(hue(@blue), saturation(@blue), lightness(@base));
  --baseline-low-check: @blue;
  --baseline-low-pill-bg: @sapphire;
  --baseline-low-pill-color: @base;
  --baseline-limited-bg: @surface0;
  --baseline-limited-engine-bg: @base;
```

**Review by uncenter — CHANGES_REQUESTED** [review]

File: styles/mdn/catppuccin.user.css [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    :root:not(.light):not(.dark) {
+      #catppuccin(@lightFlavor, @accentColor);
+    }
+  }
+
+  @media (prefers-color-scheme: dark) {
+    :root:not(.light):not(.dark) {
+      #catppuccin(@darkFlavor, @accentColor);
+    }
+  }
```
Comment (uncenter) [inline styles/mdn/catppuccin.user.css:214]:
```suggestion
  /* stylelint-disable selector-not-notation */
  @media (prefers-color-scheme: light) {
    :root:not(.light):not(.dark) {
      #catppuccin(@lightFlavor, @accentColor);
    }
  }

  @media (prefers-color-scheme: dark) {
    :root:not(.light):not(.dark) {
      #catppuccin(@darkFlavor, @accentColor);
    }
  }
  /* stylelint-enable selector-not-notation */
```

File: styles/mdn/catppuccin.user.css (hunk 3 above)
Comment (soya-daizu) [reply styles/mdn/catppuccin.user.css]: I was thinking of making custom variant of those images, but yeah we could just remove them for now.

Comment (uncenter) [reply styles/mdn/catppuccin.user.css]: Oh yeah gotcha sorry.

**Comment by uncenter** [comment]
I'd love for this theme to get merged! Is there a reason this is still a draft?

**Comment by uncenter** [comment]
Looks great! Two things I noticed:

**Color of current page in navbar**:
Screenshot 2024-02-16 at 21 24 54 (Arc)
This looks a little *too* faded to me, maybe a little closer to the color the rest of the navbar links are?

**Unthemed input field placeholder text**:
Screenshot 2024-02-16 at 21 26 12 (Arc)
(see https://developer.mozilla.org/en-US/docs/Web/CSS/::placeholder)

**Comment by soya-daizu** [comment]
> Unthemed input field placeholder text:

I can't confirm the issue on my side.
image

**Comment by uncenter** [comment]
Huh, looks the placeholder text is themed in Firefox (which I presume is what you are using) but not in Chrome/Chromium browsers.

**Review by uncenter — APPROVED** [review]
That input field placeholder thing is puzzling but doesn't seem immediately solvable and isn't very important so I'm happy to approve this :)

**Pull request merged**